main/lsp/ErrorReporter.cc

```diff
@@ -30,15 +60,26 @@ void ErrorReporter::pushDiagnostics(u4 epoch, core::FileRef file, const vector<u
 
     fileErrorStatus.lastReportedEpoch = epoch;
 
+    auto it = epochTimers.find(epoch);
+    ENFORCE(it != epochTimers.end());
+    if (!it->second.hasFirstDiagnosticEndTimes) {
+        it->second.hasFirstDiagnosticEndTimes = true;
+        for (auto &timer : it->second.firstDiagnosticLatencyTimers) {
```

Comment:
We should `setEndTime` for `lastDiagnosticLatencyTimers` here. That way, if we never send diagnostics, lastDiagnosticLatency is ~equal to firstDiagnosticLatency.